addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/handler/YamahaZoneThingHandler.java

```diff
@@ -0,0 +1,606 @@
+
+    public YamahaZoneThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        validateConfigurationParameters(configurationParameters);
+
+        // can be overridden by subclasses
+        Configuration configuration = editConfiguration();
+        for (Entry<String, Object> configurationParmeter : configurationParameters.entrySet()) {
+            configuration.put(configurationParmeter.getKey(), configurationParmeter.getValue());
+        }
+
+        updateConfiguration(configuration);
+    }
+
+    /**
+     * We handle updates of this thing ourself.
+     */
+    @Override
+    public void thingUpdated(Thing thing) {
+        this.thing = thing;
+    }
+
+    /**
+     * Calls createCommunicationObject if the host name is configured correctly.
+     */
+    @Override
+    public void initialize() {
+        // Determine the zone of this thing
+        String zoneName = (String) thing.getConfiguration().get(YamahaReceiverBindingConstants.CONFIG_ZONE);
+        if (zoneName == null) {
+            zoneName = thing.getProperties().get(YamahaReceiverBindingConstants.CONFIG_ZONE);
+        }
+        zone = zoneName != null ? ZoneControl.Zone.valueOf(zoneName) : null;
+        if (zoneName == null || zone == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Zone not set!");
+            return;
+        }
+
+        if (getBridge() != null) {
+            initializeIfBridgeAvailable();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE) {
+            initializeIfBridgeAvailable();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            zoneControl = null;
+        }
+    }
+
+    private void initializeIfBridgeAvailable() {
+        // Do nothing if zone property is missing or we are already initialised
+        if (zone == null) {
+            return;
+        }
+        if (zoneControl == null) {
+            Bridge bridge = getBridge();
+            if (bridge == null) {
+                return;
+            }
+
+            YamahaBridgeHandler brHandler = (YamahaBridgeHandler) bridge.getHandler();
+
+            zoneControl = new ZoneControl(brHandler.getCommunication(), zone, this);
+            try {
+                zoneControl.fetchAvailableInputs();
+                updateZoneInformation();
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+                return;
+            } catch (ParserConfigurationException e) {
+                logger.error(e.getLocalizedMessage());
```

Comment:
Yes, the device does not go offline if some responses cause parser errors. Some AVRs respond in a slightly different way, I want those errors to be "error" logged, because they will cause some functionality to not work.